koala/ai.ts

```diff
@@ -0,0 +1,174 @@
+import type { CoreMessage } from "@/koala/ai";
+import {
+  experimental_generateImage as generateImage,
+  generateObject,
+  generateText,
+  experimental_transcribe as transcribe,
+} from "ai";
+import { z } from "zod";
+import { errorReport } from "./error-report";
+import {
+  getDefaultImageModel,
+  getDefaultTextModel,
+  getDefaultTranscriptionModel,
+  registry,
+} from "./provider-registry";
+
+type TextModel = "fast" | "grammar" | "reasoning" | "default";
+type ImageModel = "default" | "fast";
+type LanguageModelIdentifier =
+  | `openai:${TextModel}`
+  | `anthropic:${TextModel}`;
+type ImageModelIdentifier = `openai:${ImageModel}`;
+
+export async function generateAIText(options: {
+  model?: LanguageModelIdentifier;
+  messages: CoreMessage[];
+  temperature?: number;
+  maxTokens?: number;
+}) {
+  const modelId = options.model?.includes(":")
+    ? options.model
+    : options.model
+      ? `openai:${options.model}`
+      : undefined;
+
+  const { text } = await generateText({
+    model: modelId
+      ? registry.languageModel(modelId as LanguageModelIdentifier)
+      : getDefaultTextModel(),
+    messages: options.messages,
+    temperature: 1, // options.temperature, // NOT GPT-5 COMPATIBLE TODO: Conditional temperature
+    // AI SDK v5 renamed `maxTokens` to `maxOutputTokens`.
+    maxOutputTokens: options.maxTokens,
+  });
+
+  return text;
+}
+
+export function generateStructuredOutput<S extends z.ZodTypeAny>(options: {
+  model?: string;
+  messages: CoreMessage[];
+  schema: S;
+  temperature?: number;
+  maxTokens?: number;
+}): Promise<z.infer<S>>;
+export async function generateStructuredOutput(options: {
+  model?: string;
+  messages: CoreMessage[];
+  schema: z.ZodTypeAny;
+  temperature?: number;
+  maxTokens?: number;
+}) {
+  const other = options.model ? `openai:${options.model}` : undefined;
+  const modelId = options.model?.includes(":") ? options.model : other;
+
+  const { object } = await (generateObject as any)({
+    model: modelId
+      ? registry.languageModel(modelId as LanguageModelIdentifier)
+      : getDefaultTextModel(),
+    messages: options.messages,
+    schema: options.schema,
+    temperature: 1, // options.temperature, // NOT GPT-5 COMPATIBLE TODO: Conditional temperature
+    // AI SDK v5 renamed `maxTokens` to `maxOutputTokens`.
+    maxOutputTokens: options.maxTokens,
+  });
+
+  return object;
+}
+
+export async function generateAIImage(
+  prompt: string,
+  size?: "1024x1024" | "1792x1024" | "1024x1792",
+  model?: string,
+) {
+  const other = model ? `openai:${model}` : undefined;
+  const modelId = model?.includes(":") ? model : other;
+
+  const { image } = await generateImage({
```

Comment:
So it looks like this call previously returned a URL but now returns Base64, which is causing image generation to fail:

```
TypeError: Only absolute URLs are supported
    at getNodeRequestOptions (/app/node_modules/node-fetch/lib/index.js:1327:9)
    at /app/node_modules/node-fetch/lib/index.js:1450:19
    at new Promise (<anonymous>)
    at fetch (/app/node_modules/node-fetch/lib/index.js:1447:9)
    at GCSStorageProvider.uploadFromURL (webpack-internal:///(api)/./koala/storage.ts:58:81)
    at maybeAddImageToCard (webpack-internal:///(api)/./koala/image.ts:42:65)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```